File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

//  program that encrypts messages using Caesar’s cipher

int main(int argc, string argv[])
{
  
//    
    
    if (argc != 2)
    {
        printf("Enter the key as a command-line argument\n");
        return 1;
    }
    int key = atoi(argv[1]);
    
    printf("plaintext:  ");
    string plaintext = get_string();
    
    
    int lower_alphabet[26];
    int capital_alphabet[26];
    int alpha = 97;
    
    for (int i = 0; i <= 25; i++)
    {
        lower_alphabet[i] = alpha;
        capital_alphabet[i] = alpha - 32;
        alpha++;
    }
    
    printf("ciphertext: ");
    for (int i = 0, n = strlen(plaintext);  i < n; i++)
    {
        int ciphertext;
        int digit_plaintext = (int) plaintext[i];
        if  (isalpha((int) plaintext[i])) 
        {
            if (islower((int) plaintext[i]))
            {
                ciphertext = (digit_plaintext - 97 + key);
                if (ciphertext > 25)
                {
                    ciphertext = ciphertext % 26;
                }
                
                printf("%c", lower_alphabet[ciphertext]);    
            }
            else
            {
                ciphertext = (digit_plaintext - 65 + key);
                if (ciphertext > 25)
                {
                    ciphertext = ciphertext % 26;
                }
                
                printf("%c", capital_alphabet[ciphertext]);   
            }   
        }    
        else
        {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");

    return 0;
}

File: pset6/caesar/caesar.py
# program that encrypts messages using Caesar’s cipher
import sys

def main():
    key = int(sys.argv[1])
    string_plaintext = input("plaintext:  ")
    alphabet = 'abcdefghijklmnopqrstuvwxyz'
    i = 0

    for i in range(len(string_plaintext)):
        if string_plaintext[i].isalpha():
            if string_plaintext[i].islower():
                plain = alphabet.find(string_plaintext[i])
                if plain + key > 25:
                    print(alphabet[(plain+ key) % 26], end='')
                else:
                    print(alphabet[plain + key], end='')
            else:
                plain = alphabet.find(string_plaintext[i].lower())
                if plain + key > 25:
                    print(alphabet[(plain+ key) % 26].upper(), end='')
                else:
                    print(alphabet[plain + key].upper(), end='')
        else:
            print(string_plaintext[i], end='')
    print()

if __name__ == "__main__":
    main()